### PR TITLE
refactor(console): `useConnectorInUse` hook

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

The `useConnectorInUse` hook is better to be used at the data level instead of the page level, and we need to use the `isConnectorInUse` utilities in later page refactors.

## Testing

image
image
image

### CODE REVIEW

File: packages/console/src/pages/Connectors/components/ConnectorRow/index.tsx [hunk 1]
```diff
@@ -67,14 +68,15 @@ const ConnectorRow = ({ type, connectors, onClickSetup }: Props) => {
...
-          </Status>
-        )}
-        {inUse === undefined && !connectors[0] && '-'}
+        {conditional(
+          firstConnector && (
+            <Status status={inUse ? 'enabled' : 'disabled'}>
+              {t('connectors.connector_status', {
+                context: inUse ? 'in_use' : 'not_in_use',
+              })}
+            </Status>
```
Comment (charIeszhao) [inline packages/console/src/pages/Connectors/components/ConnectorRow/index.tsx:77]: Can we just pass the `isInUse` as a prop into this `<Status />` component and let it handle the logic inside?

File: packages/console/src/pages/Connectors/components/ConnectorRow/index.tsx [hunk 2]
```diff
@@ -67,14 +68,15 @@ const ConnectorRow = ({ type, connectors, onClickSetup }: Props) => {
...
-        {inUse !== undefined && (
-          <Status status={inUse ? 'enabled' : 'disabled'}>
-            {t('connectors.connector_status', {
-              context: inUse ? 'in_use' : 'not_in_use',
-            })}
-          </Status>
-        )}
-        {inUse === undefined && !connectors[0] && '-'}
+        {conditional(
+          firstConnector && (
```
Comment (charIeszhao) [inline packages/console/src/pages/Connectors/components/ConnectorRow/index.tsx:72]: Why does only the first connector need this status?

File: packages/console/src/pages/Connectors/components/ConnectorRow/index.tsx (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/pages/Connectors/components/ConnectorRow/index.tsx:77]: The status is a common component, and it's not related to the connector.

File: packages/console/src/pages/Connectors/components/ConnectorRow/index.tsx [hunk 3]
```diff
@@ -67,14 +68,15 @@ const ConnectorRow = ({ type, connectors, onClickSetup }: Props) => {
...
       <td>
-        {inUse !== undefined && (
-          <Status status={inUse ? 'enabled' : 'disabled'}>
-            {t('connectors.connector_status', {
-              context: inUse ? 'in_use' : 'not_in_use',
-            })}
-          </Status>
-        )}
-        {inUse === undefined && !connectors[0] && '-'}
+        {conditional(
```
Comment (charIeszhao) [inline packages/console/src/pages/Connectors/components/ConnectorRow/index.tsx:71]: We don't need `conditional` in jsx

File: packages/console/src/pages/Connectors/components/ConnectorRow/index.tsx (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/pages/Connectors/components/ConnectorRow/index.tsx:72]: For multi-platform cases, there will be more than one connector.

File: packages/console/src/pages/Connectors/components/ConnectorRow/index.tsx (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/pages/Connectors/components/ConnectorRow/index.tsx:71]: This component is removed in later PR.

File: packages/console/src/hooks/use-connector-in-use.ts [hunk 4]
```diff
@@ -1,41 +1,41 @@
...
+
+      const relatedIdentifier =
+        type === ConnectorType.Email ? SignInIdentifier.Email : SignInIdentifier.Sms;
+
+      return (
+        data.signIn.methods.some(
+          ({ identifier, verificationCode }) => verificationCode && identifier === relatedIdentifier
+        ) ||
+        (data.signUp.identifiers.includes(relatedIdentifier) && data.signUp.verify)
+      );
```
Comment (charIeszhao) [inline packages/console/src/hooks/use-connector-in-use.ts:31]: I'd recommend refactor this logic, and split it into smaller pieces. For example:
```ts
const isSignInIdentifier = ...
const isSignUpIdentifier = ...

return isSignInIdentifier || isSignUpIdentifier;
```

File: packages/console/src/pages/Connectors/components/ConnectorRow/index.tsx (hunk 1 above)
Comment (charIeszhao) [reply packages/console/src/pages/Connectors/components/ConnectorRow/index.tsx:77]: ok

**Review by charIeszhao — APPROVED** [review]

File: packages/console/src/hooks/use-connector-in-use.ts (hunk 4 above)
Comment (xiaoyijun) [reply packages/console/src/hooks/use-connector-in-use.ts:31]: updated in https://github.com/logto-io/logto/pull/2813

**Pull request merged**